### ISSUE TITLE
Pegged Orders

### ISSUE THREAD

**Issue by mathsCode** [mzheravin/exchange-core#71]
Is there support for pegged order types ? or support for limit orders that dynamically move with the best market quotes ?

**Comment by mzheravin** [comment]
Actually, none of theese. 
Why do you think this is an exchange core responsibility? Imo, such orders can be managed externally. 
From performance prospective - if there are some signinficant number of orders, each GTC order that affecting BBO will instantly affect all pegged orders, right?

**Comment by mathsCode** [comment]
I think trailing and pegged orders can be part of the exchange core too. I think if a "moving" order switches from one price-level to another (e.g. a more aggressive price-level) it should sit at the back of the queue of the new price level. So it shouldn't be different from any other limit order, no ?

**Comment by mzheravin** [comment]
Well it is definitely possible to add. My concerns are: 
- Why adding it, if it can be implemented outside of the core... keeping core simple is important.
- It impacts the matching engine performance because basically each operation that affects BBO (for example any  IoC/FoK order) automatically affects same-sided pegged orders.

! Although I agree if this type of order is widely used at some certain exchange, it is good to have it in the core. 

What seems to be safe to introduce is a pedged-to-market order type.

**Comment by mathsCode** [comment]
Yes, pegged orders are popular. I'm not sure but I think there is a dedicated FIX tag for some of the pegged types as well, I think FIX 5.0 tag 835 is for the move type.
